lib/phoenix/live_dashboard/components/tab_bar_component.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Phoenix.LiveDashboard.TabBarComponent do
+  @moduledoc false
+  use Phoenix.LiveDashboard.Web, :live_component
+
+  @impl true
+  def mount(socket) do
+    {:ok, socket}
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    %{page: page, entries: entries} = assigns
+    current = current_tab(page.params, entries)
+    {:ok, assign(socket, entries: entries, current: current, page: page)}
+  end
+
+  defp current_tab(%{"tab" => tab}, entries) do
+    if Enum.any?(entries, fn {id, _} -> Atom.to_string(id) == tab end) do
+      String.to_existing_atom(tab)
+    else
+      default_tab(entries)
+    end
+  end
+
+  defp current_tab(_, entries) do
+    default_tab(entries)
+  end
+
+  defp default_tab([{id, _} | _]) do
+    id
+  end
+
+  @impl true
+  def render(assigns) do
+    ~L"""
+    <div>
+      <div class="row">
+        <div class="container">
+          <ul class="nav nav-tabs mb-4 charts-nav">
+            <%= for {id, tab} <- @entries do %>
+              <li class="nav-item">
+                <%= live_redirect(Access.fetch!(tab, :name),
+                      to: live_dashboard_path(@socket, @page, update_params(tab: id)),
+                      class: "nav-link #{if @current == id, do: "active"}") %>
+              </li>
+            <% end %>
+          </ul>
+        </div>
+      </div>
+      <%= render_content(@socket, @page, @entries, @current) %>
+    </div>
+    """
+  end
+
+  defp render_content(socket, page, entries, current) do
+    entries
+    |> Keyword.fetch!(current)
+    |> Access.fetch!(:render)
+    |> case do
+      {component, component_assigns} ->
+        component_assigns = Keyword.put(component_assigns, :page, page)
+        live_component(socket, component, component_assigns)
```

Comment:
I wrote it this way in my first version, but I changed for the following reason. In the documentation, we will say that the `page` is a required parameter for our components so it is possible the user will pass the `page` argument anyway, even if we document, in other places, that we automatically send the `page` parameter to the children components.

So if the user would use `Keyword.get_values/2` it will receive the page twice :P Using `Keyword.put/3` we solve the problem. I consider this a very remote case, so if you're ok with this I update it :D

Reply:
It will be converted to a map where the first key wins, so we are fine!